modules/data_release/ajax/AddPermission.php

```diff
@@ -182,10 +204,51 @@
     //does not currently do anything on the front-end
     //just a placeholder displaying if the operation succeeded or not
     header("Location: {$baseURL}/data_release/?addpermissionSuccess=true");
-
+} elseif ($_GET['action'] == 'getPermissions') {
+    getManagePermissionsData();
 } else {
     header("HTTP/1.1 400 Bad Request");
     echo "There was an error adding permissions";
 }
 
+/**
+ * Gets the data for the manage permission modal window.
+ *
+ * @return void
+ */
+function getManagePermissionsData(): void
+{
+    // Get permission list before any changes from main class
+    $data_release = new LORIS\data_release\data_release(
+        \Module::factory('data_release'),
+        '',
+        '',
+        '',
+        ''
+    );
 
+    $DB     = \Database::singleton();
+    $users  = $data_release->getUsersList($DB);
+    $vFiles = $data_release->getVersionedFiles($DB);
+    $prePermissions = $data_release->getUserVersionPermissions($vFiles, $DB);
+
+    $data_table_values = [];
+    $index = 0;
+    foreach ($users as $userid => $username) {
+        foreach ($vFiles as $versionName => $versionFiles) {
```

Comment:
Travis is having an issue with these two lines.

>/home/travis/build/aces/Loris/modules/data_release/ajax/AddPermission.php:237	Avoid unused local variables such as '$userid'.
/home/travis/build/aces/Loris/modules/data_release/ajax/AddPermission.php:238	Avoid unused local variables such as '$versionFiles'.

Since you're not using `$userid`, you can change the first loop to be:
`foreach ($users as $username) {`
PHP iterates over the values of an array by default.

For the next one you're not using `$versionFiles`. In this case you can just iterate over the keys of the array:
`foreach (array_keys($vFiles) as $versionName) {`

Reply:
@johnsaigle Thank you! Resolved in the last commit.
I am a bit confused though because I thought Travis failed when running npm which is for JS file, no?
Anyway, thanks for your help!

Reply:
`npm` is usually associate with JS but we have it configured to run other scripts for us.

If you look in `package.json` you'll see references to things like `npm run tests:unit` which in turn actually point to PHP or bash scripts. 

I'm not exactly sure why we do this but... that's how it is.